src/gt4py/backend/gtc_backend/dace/backend.py

```diff
@@ -0,0 +1,389 @@
+            dimensions = array_dimensions(array)
+            domain_ndim = sum(dimensions)
+            data_ndim = len(array.shape) - domain_ndim
+            sid_def = """gt::as_{sid_type}<{dtype}, {num_dims},
+                std::integral_constant<int, {unique_index}>>({name})""".format(
+                sid_type="cuda_sid"
+                if array.storage in [dace.StorageType.GPU_Global, dace.StorageType.GPU_Shared]
+                else "sid",
+                name=name,
+                dtype=array.dtype.ctype,
+                unique_index=self.unique_index(),
+                num_dims=len(array.shape),
+            )
+            if domain_ndim != 3:
+                gt_dims = [
+                    f"gt::stencil::dim::{dim}"
+                    for dim in "ijk"
+                    if any(
+                        dace.symbolic.pystr_to_symbolic(f"__{dim.upper()}") in s.free_symbols
+                        for s in array.shape
+                        if hasattr(s, "free_symbols")
+                    )
+                ]
+                if data_ndim:
+                    gt_dims += [
+                        f"gt::integral_constant<int, {3 + dim}>" for dim in range(data_ndim)
+                    ]
+                sid_def = "gt::sid::rename_numbered_dimensions<{gt_dims}>({sid_def})".format(
+                    gt_dims=", ".join(gt_dims), sid_def=sid_def
+                )
+
+            res.append(
+                "gt::sid::shift_sid_origin({sid_def}, {name}_origin)".format(
+                    sid_def=sid_def, name=name
+                )
+            )
+        # pass scalar parameters as variables
+        for name in (n for n in sdfg.symbols.keys() if not n.startswith("__")):
+            res.append(name)
+        return res
+
+    def generate_sdfg_bindings(self, gtir, sdfg, module_name):
+
+        return self.mako_template.render_values(
+            name=sdfg.name,
+            module_name=module_name,
+            entry_params=self.generate_entry_params(gtir, sdfg),
+            sid_params=self.generate_sid_params(sdfg),
+        )
+
+    @classmethod
+    def apply(cls, gtir: gtir.Stencil, sdfg: dace.SDFG, module_name: str) -> str:
+        generated_code = cls().generate_sdfg_bindings(gtir, sdfg, module_name=module_name)
+        formatted_code = codegen.format_source("cpp", generated_code, style="LLVM")
+        return formatted_code
+
+
+@gt_backend.register
+class GTCDaceBackend(BaseGTBackend, CLIBackendMixin):
+    """DaCe python backend using gtc."""
+
+    name = "gtc:dace"
+    GT_BACKEND_T = "dace"
+    languages = {"computation": "c++", "bindings": ["python"]}
+    storage_info = {
+        "alignment": 1,
+        "device": "cpu",
+        "layout_map": make_x86_layout_map,
+        "is_compatible_layout": x86_is_compatible_layout,
+        "is_compatible_type": x86_is_compatible_layout,
+    }
+
+    options = BaseGTBackend.GT_BACKEND_OPTS
+    PYEXT_GENERATOR_CLASS = GTCDaCeExtGenerator  # type: ignore
```

Comment:
```suggestion
    PYEXT_GENERATOR_CLASS = GTCDaCeExtGenerator  # type: ignore
    USE_LEGACY_TOOLCHAIN = False
```

Reply:
Glad you had the opportunity to return the favor :sweat_smile: